### PR TITLE
Fixing #3232, crash when making a server push

### PR DESCRIPTION
**Pull request by zizhong**
@masaori335 @maskit please review when you get a chance!

### CODE REVIEW

**Comment by zwoop** [comment]
[approve ci].

**Comment by zwoop** [comment]
[add to whitelist]

**Comment by zwoop** [comment]
[approve ci]

**Comment by zwoop** [comment]
[approve ci]
[add to whitelist]

File: proxy/InkAPI.cc [hunk 1]
```diff
@@ -7748,7 +7748,7 @@ TSHttpTxnServerPush(TSHttpTxn txnp, const char *url, int url_len)
   Http2Stream *stream = dynamic_cast<Http2Stream *>(sm->ua_txn);
   if (stream) {
     Http2ClientSession *parent = static_cast<Http2ClientSession *>(stream->get_parent());
-    if (!parent->is_url_pushed(url, url_len)) {
+    if (parent && !parent->connection_state.is_state_closed() && !parent->is_url_pushed(url, url_len)) {
```
Comment (masaori335) [inline proxy/InkAPI.cc]: Should the mutex of parent or connection_state be locked?

Comment (zizhong) [reply proxy/InkAPI.cc]: yes, good catch!

**Comment by zizhong** [comment]
@masaori335 locking added

**Comment by maskit** [comment]
@zwoop I think this is a 7.x candidate.

**Comment by zwoop** [comment]
Cherry-picked to 7.1.x

**Pull request merged**